tests/unit/testplan/testing/multitest/driver/test_tcp_server.py

```diff
@@ -38,6 +39,46 @@ def tcp_client(tcp_server, runpath_module):
         yield client
 
 
+@pytest.fixture(scope="module")
+def tcp_server_port_aliased(runpath_module):
+    """Start and yield a TCP server driver using port aliasing."""
+    env = Environment()
+
+    def get_mocked_port_name():
+        for port in range(2000, 65535):
+            try:
+                return socket.getservbyport(port)
```

Comment:
nothing guaranties that on the testing hosts there are port names configured. better to patch port_to_int to return what you expect

Reply:
Yeah 😊. It looks a bit hacky. I was thinking of how to create a port alias at runtime but that would demand root access so I found this new way which worked for multiple environments given a range of [**un-privileged ports.**](https://www.w3.org/Daemon/User/Installation/PrivilegedPorts.html) I would be glad if you could help me get a better way around it.

Reply:
If I were you, I would write a helper to resolve port (being it int, context, value or string) use that in both client and server, and write a pure unittest for that function, with patching the port_to_int (https://docs.python.org/3/library/unittest.mock.html#unittest.mock.patch) to return the desired port for the portname the test test is using. I would not do any server startup, and magic to figure out if named ports exists, not talking about the possibility that the named port you pick might be already taken and your test is just failing. This is a nice way of adding flaky tests.

Reply:
Thanks for the feedback. I will do the necessary refinements.